CONTRIBUTING.md

```diff
@@ -0,0 +1,111 @@
+
+```ascii
+services/
+└── database/
+    └── relational/
+    └── warehouse/
+```
+
+Each resource should contain the following files in `.yaml` format:
+
+- `controls.yaml`: The common compliance controls for the resource.
+- `features.yaml`: The common features which define the resource.
+- `threats.yaml`: The security threats related to the resource.
+- `metadata.yaml`: Metadata on the resource such as resource details.
+
+The resource should also contain a `tests` directory containing details on how to test the resource is compliant.
+
+The top level of the `services` directory contains `yaml` files for common controls, features and threats which are common to multiple resources. It also contains the [`service-families.yaml`](/services/service-families.yaml) file which documents the service families and their descriptions.
+
+## VSCode
+
+### Snippets
+
+[VSCode snippets](https://code.visualstudio.com/docs/editor/userdefinedsnippets) are defined for common [features](.vscode/common-features.code-snippets), [threats](.vscode/common-threats.code-snippets) and [controls](.vscode/common-controls.code-snippets).
+To make use of these snippets start typing the snippet prefix in any `yaml` file and VSCode will offer you the option to auto-complete the snippet e.g.
+
+```yaml
+CF3
+```
+
+![snippet](/docs/resources/snippet.gif)
+
+#### Snippet overview
+
+| Snippet           | Description                            | File                                                                   |
+| ----------------- | -------------------------------------- | ---------------------------------------------------------------------- |
+| Threats File      | Create a blank FINOS CCC Threats file  | [threats.code-snippets](.vscode/threats.code-snippets)                 |
+| Threat            | Create a blank FINOS CCC Threat        | [threats.code-snippets](.vscode/threats.code-snippets)                 |
+| Common Threats\*  | Create a common FINOS CCC Threat       | [common-threats.code-snippets](.vscode/common-threats.code-snippets)   |
+| Features File     | Create a blank FINOS CCC Features file | [features.code-snippets](.vscode/features.code-snippets)               |
+| Feature           | Create a blank FINOS CCC Feature       | [features.code-snippets](.vscode/features.code-snippets)               |
+| Common Features\* | Create a common FINOS CCC Feature      | [common-features.code-snippets](.vscode/common-features.code-snippets) |
+| Controls File     | Create a blank FINOS CCC Controls file | [controls.code-snippets](.vscode/controls.code-snippets)               |
+| Control           | Create a blank FINOS CCC Control       | [controls.code-snippets](.vscode/controls.code-snippets)               |
+| Common Controls\* | Create a common FINOS CCC Control      | [common-controls.code-snippets](.vscode/common-controls.code-snippets) |
```

Comment:
We need to add the metadata option.

We should add a column with the prompts to start typing to get the snippet up?

Reply:
I'll add the metadata one, missed that! 
For the prompts, feels like that's just duplicating the info in the linked snippet file? It's also not clear what we'd put for the more general ones e.g. `Common Features*`?

Reply:
it would be just so that pepole don't have to go into the files and parse the info from the snippets.

for common ones the prompt would be e.g.

cf
ct
cc

I can add these later though if I get time. It all makes sense in my head, maybe not so much when I write it down in the comments here :)

Reply:
Ok, I've update that - I suspect our prompts for the common controls are wrong though (CT rather than CC)?

Reply:
That was fixed, just checking where it went...

Reply:
Apparently it wasn't fixed, it should have been part of this: https://github.com/finos/common-cloud-controls/commit/03264fc9f1a4b3703cbddf64dd55e33c645dd1e4

Reply:
😀 - I can add it in here if you like?

Reply:
If you have the time then yeah go for it